src/types.rs

```diff
@@ -616,6 +616,11 @@ pub enum AuthorizationStatus {
 #[serde(tag = "type", content = "value", rename_all = "camelCase")]
 pub enum Identifier {
     Dns(String),
+
+    /// An IP address (IPv4 or IPv6) identifier
+    ///
+    /// Note that not all ACME servers will accept an order with an IP address identifier.
+    Ip(String),
```

Comment:
How about we just store an IpAddr in there?

Reply:
Done.